app/views/usergroups/_form.html.erb

```diff
@@ -48,7 +48,8 @@
                 <td><%= display_link_if_authorized(_('Refresh'),
                                                          hash_for_refresh_external_usergroup_path(:id => usergroup),
                                                          :method => :put, :id => usergroup.id,
-                                                         :class => "btn btn-default", :title => _('Synchronize group from authentication source')) %>
+                                                         :class => "btn btn-default", :title => _('Synchronize group from authentication source'),
+                                                         :disabled => usergroup.auth_source_external?) %>
```

Comment:
Nit: A better API would be `usergroup.supports_refresh?`. And then something like `auth_source.supports_refresh?`. The superclass of AuthSourceExternal would then have a method that returns false and AuthSourceExternal would override this and return true. This is much more extensible this way.

Reply:
should we display the button at all if the auth source doesn't support it?

Reply:
I think it's nicer to have a disabled button or at least some text explaining, why it can't get synced.

Reply:
Yeah, something like „Sync is not available for external auth sources”. I think we should try to explain more in the application rather than forcing users to read the manual.